src/main/java/com/smoope/utils/traverson/Traverson.java

```diff
@@ -242,53 +251,51 @@ private CallResult call(final RequestMethod method) throws TraversonException, I
         }
 
         private TraversingResult traverseToFinalUrl() throws TraversonException, IOException {
-            return getAndFindLinkWithRel(rootUri, rels.iterator());
+            return getAndFindLinkWithRel(TraversingResult.url(rootUri), rels.iterator());
         }
 
-        private TraversingResult getAndFindLinkWithRel(String url, Iterator<String> rels)
+        private TraversingResult getAndFindLinkWithRel(TraversingResult result, Iterator<String> rels)
                 throws TraversonException, IOException {
-            log.debug("Traversing an URL: {}", url);
+            log.debug("Traversing: {}", result);
 
             if (!rels.hasNext()) {
-                return TraversingResult.url(url);
+                return result;
             }
 
-            TraversonResult<Object> response = prepareResponse(
+            TraversonResult<Object> response = result.isUrl()
+                ? prepareResponse(
                     CallResult.response(
-                            handleErrors(
-                                    client.newCall(
-                                            prepareRequest(url, RequestMethod.GET)
-                                    ).execute()
-                            )
+                        handleErrors(
+                            client.newCall(
+                                prepareRequest(result.getUrl(), GET)
+                            ).execute()
+                        )
                     ),
                     TraversonResult.class
-            );
+                )
+                : serializer.fromJson(serializer.toJson(result.getEmbedded()), TraversonResult.class);
```

Comment:
@systemfreund It's kinda messy but couldn't find another way of supporting "external" resources. Main issue was that in case of request to embedded resource directly, we have to respond with the resource itself. Otherwise we need grab relations from it.